server/controllers/user.js

```diff
@@ -0,0 +1,153 @@
+// import modules
+import jwt from 'jsonwebtoken';
+import bcrypt from 'bcrypt-nodejs';
+import models from '../models';
+
+require('dotenv').config();
+
+// create reference to db model
+const User = models.User;
+
+// create variable to hold environmental variable SECRET
+const secret = process.env.SECRET;
+
+export default {
+
+  /**
+     * @description creates a new user controller
+     * @method
+     * @param {Object} req - Request object
+     * @param {Object} res - Response object
+     * @returns {Object} json - payload
+     */
+  signup(req, res) {
+    User
+      .find({ where: { email: req.body.email } })
+      .then((response) => {
+        if (response) {
+          return res.status(409).send({
+            message: 'User already exists.',
+          });
+        }
+        User
+          .create({
+            firstName: req.body.firstName,
+            lastName: req.body.lastName,
+            email: req.body.email,
+            password: req.body.password,
+            imageUrl: req.body.imageUrl,
+          })
+          .then((user) => {
+            const userData = { user };
+            const token = jwt.sign(userData, secret);
+            return res.status(201).send({
+              message: 'Registered successfully',
+              token,
+            });
+          });
+      })
+      .catch(error => res.status(400).send({ message: error.message }));
+  },
+
+  /**
+     * @description sign in user controller
+     * @method
+     * @param {Object} req - Request object
+     * @param {Object} res - Response object
+     * @returns {Object} json - payload
+     */
+  signin(req, res) {
+    User
+      .find({ where: { email: req.body.email } })
+      .then((user) => {
+        if (!user) {
+          return res.status(404).send({ message: 'User is not registered!.' });
+        }
+
+        if (bcrypt.compareSync(req.body.password, user.password)) {
+          const userData = { user };
+          const token = jwt.sign(userData, secret);
+          return res.status(200).send({
+            message: 'Logged in Successfully',
+            token,
+          });
+        }
+        return res.status(409).send({
```

Comment:
409 Status code implies there is a conflict and the user cannot resubmit the correct request. That's not the case here. A 401 would be more appropriate here.

http://www.restapitutorial.com/httpstatuscodes.html

Click on each status code to see detailed explanation.

Reply:
Am checking if the user already exists